## Extreme value analysis using univariate methods

In [12]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
from pylab import rcParams

In [13]:
address = '../data/iris.data.csv'
df= pd.read_csv(filepath_or_buffer=address, header=None, sep=',')

df.columns = ['Sepal Length', 'Sepal Width', 'Petal Length', 'Petal Width', 'Species']
x = df.iloc[:,0:4].values
y = df.iloc[:,4].values
df[:5]

Sepal Length  Sepal Width  Petal Length  Petal Width Species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

### Identifying outliers from Tukey boxplots

### Applying Tukey outlier labeling

In [14]:
pd.options.display.float_format = '{:.1f}'.format
X_df = pd.DataFrame(x)
print(X_df.describe())

          0     1     2     3
count 150.0 150.0 150.0 150.0
mean    5.8   3.1   3.8   1.2
std     0.8   0.4   1.8   0.8
min     4.3   2.0   1.0   0.1
25%     5.1   2.8   1.6   0.3
50%     5.8   3.0   4.3   1.3
75%     6.4   3.3   5.1   1.8
max     7.9   4.4   6.9   2.5


Detect outliers arithmetically 

To detect outliers, check if any points are below Q1 - 1.5IQR or above Q3 + 1.5IQR. In column 3, the IQR is calculated as Q3 - Q1 = 3.3 - 2.8 = 0.5. Multiplying the IQR by 1.5 gives 0.75, so the lower bound is 2.8 - 0.75 = 2.05, which is higher than our minimum value of 2.0.